services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/WriteBatch.java

```diff
@@ -226,13 +222,12 @@ public Builder<T> addDeleteItem(Key key) {
 
         @Override
         public Builder<T> addDeleteItem(T keyItem) {
```

Comment:
```java
WriteBatch.builder(dbBeanClass)
   .addDeleteItem(putItem)
   .mappedTableResource(table)
   .build())
```

We will see the issue in the above case too with the current fix.

I think the right fix is collect the DeleteItem/PutItemRequest and then at the end do the null check in the build()

Or another option which is not backward compatible is arrange the chaining of Builders such that `mappedTableResource` always comes first

Reply:
This already throws NPE today if `mappedTableResource` is null because we call `mappedTableResource.keyFrom` on the line below. This change basically adds a better message to the exception in this case

Reply:
Yeap I guess https://github.com/aws/aws-sdk-java-v2/pull/4267 which was merges couple of weeks ago.
This added the check , I thing we need to rework or revert that PR

Reply:
Got it , Thanks